src/adios4dolfinx/adios2_helpers.py

```diff
@@ -52,17 +52,20 @@ def ADIOSFile(
     engine: str,
     mode: adios2.Mode,
     io_name: str,
+    comm: MPI.Intracomm | None = None,
 ):
     io = adios.DeclareIO(io_name)
     io.SetEngine(engine)
     # ADIOS2 sometimes struggles with existing files/folders it should overwrite
     if mode == adios2.Mode.Write:
         filename = Path(filename)
-        if filename.exists():
+        if filename.exists() and comm is not None and comm.rank == 0:
             if filename.is_dir():
                 shutil.rmtree(filename)
             else:
                 filename.unlink()
```

Comment:
Another option here (which I usually do) is to pass
```python
shutil.rmtree(filename, ignore_errors=True)
```
and
```python
filename.unlink(missing_ok=True)
```
That way you don't need to check for communicator and rank (but this is also OK). Having the barrier at the end is probably a good idea anyway (we don't want some processes to start writing while others are deleting).

Reply:
As we do want the barrier to ensure that a process doesn't delete while another writes, we need the comm input.
I believe this is the best (and most explicit way) of dealing with it. If anything happens during deletion with shutil.rmtree, I'd rather like the user to see the error.

Reply:
As you said, we don't want processes to delete while others have started writing. So we need a barrier, and a communicator for that. And then since we have a communicator anyway, better checking the rank and making sure only one processor is deleting while other are waiting. (`ignore_errors=True` might hide other errors)